### ISSUE TITLE
[clang-tidy] False positive misc-const-correctness pointer used by placement new

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#187012]
Example:

```cpp
#include <new>

void f()
{
    alignas(int) unsigned char x[sizeof(int)];
    unsigned char* const px = &x[0];
    new(px) int{123};
}
```
https://godbolt.org/z/E9cTKn148

Here, making the pointee `const` is not possible, since it's written to by `operator new`.
